csm/core/services/security.py

```diff
@@ -350,3 +350,37 @@ async def check_certificate_expiry_time_task(self):
                            tzinfo=timezone.utc),
             interval=timedelta(days=1)
         )
+
+    async def _get_name_details(self, name):
+        """
+        Get x509 Name object details (i.e Subject and Issuer)
+        """
+        name_details = {}
+        for k,v in x509.oid._OID_NAMES.items():
+            try:
+                name_details[v] = name.get_attributes_for_oid(k)[0].value
+            except Exception as e:
+                Log.error(f"No value for {v}:{e}")
+        return name_details
+
+    async def get_certificate_details(self):
+        """
+        Get current certificate details
+        """
+        path = Conf.get(const.CSM_GLOBAL_INDEX, "HTTPS.certificate_path")
+        def load():
+            return self._load_certificate(path)
+        cert = await self._loop.run_in_executor(self._executor, load)
```

Comment:
Why `self._loop.run_in_executor` is required here? You can simply call the load certificate method.

Reply:
Done.